packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -846,6 +846,21 @@ const UpgradeCreditHelpIconLink = () => {
 	);
 };
 
+function FetchOverrideCode( { product }: { product: ResponseCartProduct } ) {
+	//Ensure product has cost overrides
+	if ( ! product || ! Array.isArray( product.cost_overrides ) ) {
+		return '';
+	}
+
+	// Find the first cost override with a non-empty 'human_readable_reason'
+	const firstValidOverride = product.cost_overrides.find(
```

Comment:
Off the top of my head I can't think of any, but, are there situations where multiple cost overrides are applied to a product? And if so, would we want to ensure specific overrides are applied rather than the first of the group?

Reply:
@sirbrillig In a situation with multiple overrides, how do we select which override code is used? I am also assuming multiple override codes can't be displayed as that will not look clean.

Reply:
In a situation with multiple overrides, all the overrides are used, in the order they were applied. For example, if a 10% discount and a 50% discount are applied to a $100 product, then it will be $45 (10% of 100 is 90, and 50% of 90 is 45).

An example of this is when we have a multi-year domain in the cart when a bundled domain credit is available and a coupon code is applied. The bundled domain credit will make the first year free, and the coupon will apply to the other years.

> I am also assuming multiple override codes can't be displayed as that will not look clean.

I agree. I had tried to do this in https://github.com/Automattic/wp-calypso/pull/66029 but it didn't look very good. We've come up with a new design to solve this in the checkout redesign project (https://github.com/Automattic/payments-shilling/issues/1969):

<img width="267" alt="286438404-94db31ab-7cb0-4065-826e-c5e54c90d085" src="https://github.com/Automattic/wp-calypso/assets/2036909/1e25a848-552b-4a69-be10-2748d48ca854">

So for this current design, it might be a good idea just to fix the bug ("upgrade credits" being shown for all discounts) without showing the reason for other discounts. We know that the code for upgrade credits is `recent-plan-proration`, so we could hide the message if that code is not present in the list of overrides.